internal/impl/kafka/enterprise/integration_test.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed as a Redpanda Enterprise file under the Redpanda Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// https://github.com/redpanda-data/connect/blob/main/licenses/rcl.md
+
+package enterprise_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/redpanda-data/benthos/v4/public/service/integration"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/redpanda-data/connect/v4/internal/impl/kafka/enterprise"
+	"github.com/redpanda-data/connect/v4/internal/protoconnect"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kerr"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/kmsg"
+)
+
+func createKafkaTopic(ctx context.Context, address, id string, partitions int32) error {
+	topicName := fmt.Sprintf("topic-%v", id)
+
+	cl, err := kgo.NewClient(kgo.SeedBrokers(address))
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	createTopicsReq := kmsg.NewPtrCreateTopicsRequest()
+	topicReq := kmsg.NewCreateTopicsRequestTopic()
+	topicReq.NumPartitions = partitions
+	topicReq.Topic = topicName
+	topicReq.ReplicationFactor = 1
+	createTopicsReq.Topics = append(createTopicsReq.Topics, topicReq)
+
+	res, err := createTopicsReq.RequestWith(ctx, cl)
+	if err != nil {
+		return err
+	}
+	if len(res.Topics) != 1 {
+		return fmt.Errorf("expected one topic in response, saw %d", len(res.Topics))
+	}
+	return kerr.ErrorForCode(res.Topics[0].ErrorCode)
+}
+
+func readNKafkaMessages(ctx context.Context, t testing.TB, address, topic string, nMessages int) (res []string) {
+	t.Helper()
+
+	cl, err := kgo.NewClient(
+		kgo.SeedBrokers(address),
+		kgo.ClientID("meow"),
+		kgo.ConsumeTopics(topic),
+	)
+	require.NoError(t, err)
+
+	defer cl.Close()
+
+	for len(res) < nMessages {
+		fetches := cl.PollRecords(ctx, nMessages-len(res))
+		require.NoError(t, ctx.Err(), len(res))
+		fetches.EachError(func(s string, i int32, err error) {
+			t.Error(err)
+		})
+		fetches.EachRecord(func(r *kgo.Record) {
+			res = append(res, string(r.Value))
+		})
+	}
+	return
+}
+
+func TestIntegration(t *testing.T) {
+	integration.CheckSkip(t)
+	t.Parallel()
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	kafkaPort, err := integration.GetFreePort()
+	require.NoError(t, err)
+
+	kafkaPortStr := strconv.Itoa(kafkaPort)
+	options := &dockertest.RunOptions{
```

Comment:
FYI, there's a redpanda testcontainer.
The interface is very simple.

```go
ctx := context.Background()

redpandaContainer, err := redpanda.Run(ctx,
    "docker.redpanda.com/redpandadata/redpanda:v23.3.3",
    redpanda.WithEnableSASL(),
    redpanda.WithEnableKafkaAuthorization(),
    redpanda.WithEnableWasmTransform(),
    redpanda.WithNewServiceAccount("superuser-1", "test"),
    redpanda.WithNewServiceAccount("superuser-2", "test"),
    redpanda.WithNewServiceAccount("no-superuser", "test"),
    redpanda.WithSuperusers("superuser-1", "superuser-2"),
    redpanda.WithEnableSchemaRegistryHTTPBasicAuth(),
)
if err != nil {
    log.Fatalf("failed to start container: %s", err)
}

// Clean up the container
defer func() {
    if err := redpandaContainer.Terminate(ctx); err != nil {
        log.Fatalf("failed to terminate container: %s", err)
    }
}()
```

https://golang.testcontainers.org/modules/redpanda/